### PR TITLE
FLW-644 iOS: Add MoonPay support for Buy token feature

### PR DESCRIPTION
**Pull request by stas-litvinov**
Integrate MoonPay provider for buying DOT tokens

### CODE REVIEW

File: fearless/Common/PurchaseProvider/MoonpayProvider.swift [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        self.colorCode = colorCode
+        return self
+    }
+
+    func with(callbackUrl: URL) -> Self {
+        self.callbackUrl = callbackUrl
+        return self
+    }
+
+    func buildPurchaseAction(
```
Comment (ef1rspb) [inline fearless/Common/PurchaseProvider/MoonpayProvider.swift:29]: `buildPurchaseAction(s)`?

File: fearless/Common/PurchaseProvider/MoonpayProvider.swift [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        for chain: Chain,
+        assetId _: WalletAssetId?,
+        address: String
+    ) -> [PurchaseAction] {
+        let optionUrl: URL?
+
+        switch chain {
```
Comment (ef1rspb) [inline fearless/Common/PurchaseProvider/MoonpayProvider.swift:36]:
```swift
guard chain == .polkadot else { return [] }
```

File: fearless/Common/PurchaseProvider/MoonpayProvider.swift [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        case .polkadot:
+            optionUrl = buildURLForToken("DOT", address: address)
+        case .kusama:
+            optionUrl = nil
+        case .westend:
+            optionUrl = nil
+        }
+
+        if let url = optionUrl {
+            let action = PurchaseAction(title: "MoonPay", url: url, icon: R.image.iconMoonPay()!)
```
Comment (ef1rspb) [inline fearless/Common/PurchaseProvider/MoonpayProvider.swift:46]: force unwrap icon. Let icon be an optional value

Comment (stas-litvinov) [reply fearless/Common/PurchaseProvider/MoonpayProvider.swift:46]: It's guaranteed that icon is always there. If it's not, the project will not build

File: fearless/Common/PurchaseProvider/MoonpayProvider.swift (hunk 2 above)
Comment (stas-litvinov) [reply fearless/Common/PurchaseProvider/MoonpayProvider.swift]: fixed

**Review by ef1rspb — APPROVED** [review]

**Pull request merged**